ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/ClasspathComputer.java

```diff
@@ -108,6 +109,7 @@ private static IClasspathEntry[] collectInOriginalOrder(List<IClasspathEntry> or
 		});
 		// using the order of reloading, append new entries (in the map still)
 		result.addAll(resultingReloadedByPath.values());
+		result.replaceAll(e -> updateTestAttribute(isTestPlugin, e));
```

Comment:
> If I understand correctly, this would remove each TEST attribute from CPE_SOURCE entries when the plugin name isn't matching the pattern, no?

That's correct.

> Shouldn't we just set the TEST attribute when indicated by the pattern, but leave alone when not indicated?

That is something that can be discussed.


> And why does updateTestAttribute() touch CPE_SOURCE entries only, but not CPE_LIBRARY (or similar, if any?) entries referring to bundle-local content?

This probably used to be like this since PDE does not (yet) have the concept of test-dependencies, like for example Maven. IIRC the `TEST` flag was mainly introduced for the Maven use-case where you can have dependencies that should only be available for test-src-folders. For now in PDE the test-src-folders only prevent usage of test-src in non-test-sources.